src/pharmpy/tools/external/nonmem/results_file.py

```diff
@@ -9,6 +10,41 @@
 from pharmpy.deps import pandas as pd
 from pharmpy.model.external.nonmem.dataset.nmtran import IOFromChunks
 
+TAG = re.compile(r' #([A-Z]{4}):\s*(.*)')
+END_TERE = re.compile(r'(0|1)')  # The part we need after #TERE: will precede the next ^1 or ^0
+CLEANUP = re.compile(r'\*+\s*')
+
+
+def _decode_lst(line: bytes):
+    # Since lst-files sometimes can have mixed encodings:
+    # Allow for separate encodings of date strings and the rest of the content
+    # Always try utf-8 first and fallback to latin-1 separately for each section
+    try:
+        row = line.decode('utf-8')
+    except UnicodeDecodeError:
+        row = line.decode('latin-1', errors='ignore')
+
+    pos = len(row) - 2
+    if pos >= 0 and row[pos] == '\r':
+        return row[:pos] + '\n'
+    else:
+        return row
```

Comment:
Almost half of the time is spent checking for `\r`, even if there are none. Maybe there is a better way?

```suggestion
    try:
        return line.decode('utf-8')
    except UnicodeDecodeError:
        return line.decode('latin-1', errors='ignore')
```

**BEFORE**

<img width="1461" height="317" alt="Screenshot 2025-12-02 at 17 01 56" src="https://github.com/user-attachments/assets/ee996c9a-9ecb-4be2-9185-e4280c8e299a" />


**AFTER**

<img width="1457" height="315" alt="Screenshot 2025-12-02 at 17 08 58" src="https://github.com/user-attachments/assets/e615fc88-f6a4-4936-a2c7-6ad7fc1d01a4" />

Reply:
This will not be a problem anymore if we implement https://github.com/pharmpy/pharmpy/pull/4186#discussion_r2583986041 because the bottleneck is detecting specific line start sequences which are encoded similarly in `utf8` and `latin1`: `b" #"` and (maybe) `b" *"`.